pulser/simulation/simresults.py

```diff
@@ -112,13 +118,14 @@ def get_final_state(self, reduce_to_basis=None, ignore_global_phase=True,
 
         return final_state.tidyup()
 
-    def expect(self, obs_list):
+    def expect(self, obs_list: list[Union[qutip, np.ndarray]]
```

Comment:
The typing of `obs_list` is not right, it's rather: 
```suggestion
    def expect(self, obs_list: list[Union[qutip.QObj, ArrayLike]]
```
I can see that you got the wrong idea from the docstring, you should change it too